### DISCUSSION TITLE
docker一键启动遇到的问题

### DISCUSSION THREAD

**Discussion by kyrieirvingcsgo** [OpenCSGs/csghub#614] [General]
1
![image](https://github.com/user-attachments/assets/d49cd83d-75ff-4bb7-b6c3-f9d2984773fe)
![image](https://github.com/user-attachments/assets/aac65281-9044-4448-999d-6c8efa74e48e)
执行.startup.sh 提示 脚本中有错误语法  **docker compose -f**   应该为 **docker-compose -f**
![image](https://github.com/user-attachments/assets/12bf7339-e959-4ff2-bc60-42b42a480ec9)

最后
![image](https://github.com/user-attachments/assets/fa39b4ac-93fb-4b14-8d64-feddbdeb317e)
3001 gitea  和  登录的31001没有起来

![image](https://github.com/user-attachments/assets/84717452-db39-4c9f-b1e1-6399b22b7781)
然后执行了
docker-compose start gitea  
![image](https://github.com/user-attachments/assets/018dce2b-ff0d-43fa-bd60-d4d276cd591d)
但最终还没有起来

gitdata目录为空，gitlog下面的boot.log文件内容如下
![image](https://github.com/user-attachments/assets/85d34672-cf31-4d15-a45f-84f71ebf0fbb)

--------------------------------------------
![image](https://github.com/user-attachments/assets/31ebf73c-4e4c-4d6f-9d5f-f7a2095eae3e)
8be8665541e0   opencsg-registry.cn-beijing.cr.aliyuncs.com/opencsg_public/csghub_server:latest 的logs

```
Error: failed to init router: error creating user handler:failed to create user component: failed to create git server,error:Post "http://gitea:3001/api/v1/users/root/tokens": dial tcp: lookup gitea on 127.0.0.11:53: server misbehaving
init logger, level: INFO, format: json
{"time":"2024-09-12T07:22:56.01008769Z","level":"ERROR","msg":"Error sending request:","!BADKEY":"Post \"http://gitea:3001/api/v1/users/root/tokens\": dial tcp: lookup gitea on 127.0.0.11:53: server misbehaving"}
{"time":"2024-09-12T07:22:56.010202869Z","level":"ERROR","msg":"Fail to create git server access token","error: ":"Post \"http://gitea:3001/api/v1/users/root/tokens\": dial tcp: lookup gitea on 127.0.0.11:53: server misbehaving"}
{"time":"2024-09-12T07:22:56.010210034Z","level":"ERROR","msg":"Failed to find or create token","error: ":"Post \"http://gitea:3001/api/v1/users/root/tokens\": dial tcp: lookup gitea on 127.0.0.11:53: server misbehaving"}
Error: failed to init router: error creating user handler:failed to create user component: failed to create git server,error:Post "http://gitea:3001/api/v1/users/root/tokens": dial tcp: lookup gitea on 127.0.0.11:53: server misbehaving
init logger, level: INFO, format: json
{"time":"2024-09-12T07:23:56.324703235Z","level":"ERROR","msg":"Error decoding JSON response:","!BADKEY":"invalid character 'N' looking for beginning of value"}
{"time":"2024-09-12T07:23:56.324821608Z","level":"ERROR","msg":"Fail to create git server access token","error: ":"invalid character 'N' looking for beginning of value"}
{"time":"2024-09-12T07:23:56.324827735Z","level":"ERROR","msg":"Failed to find or create token","error: ":"invalid character 'N' looking for beginning of value"}
Error: failed to init router: error creating user handler:failed to create user component: failed to create git server,error:invalid character 'N' looking for beginning of value
```

![image](https://github.com/user-attachments/assets/3bd43586-684f-44a4-8280-a347d087f832)
docker的gitea服务日志

**Comment by ganisback** [comment]
先start user_server, 再启动gitee 完了之后restart nginx

**Reply by kyrieirvingcsgo** [reply]
> 先start user_server, 再启动gitee 完了之后restart nginx
![image](https://github.com/user-attachments/assets/c284e7bf-ebb2-41db-a3e2-857ca8aa3ddf)
![image](https://github.com/user-attachments/assets/1ec06a24-a73a-4630-8941-348c655ddd21)
http://121.37.3.254:3000/datasets

大佬该执行都执行了

**Comment by Rader** [comment]
@kyrieirvingcsgo 安装的哪个版本？

**Reply by kyrieirvingcsgo** [reply]
> @kyrieirvingcsgo 安装的哪个版本？

v8版本

**Comment by HaiHui886** [comment]
报503错误，上传一下all_in_one_nginx_1和all_in_one_csghub_server_1的container日志

**Reply by kyrieirvingcsgo** [reply]
> container日志

[all-in-one-nginx-container.log](https://github.com/user-attachments/files/16976487/all-in-one-nginx-container.log)
[all_in_one_csghub_server_1.log](https://github.com/user-attachments/files/16976513/all_in_one_csghub_server_1.log)
大佬，文件都在这了

**Comment by HaiHui886** [comment]
nginx的log报错：{"log":"nginx: [emerg] server name \"*.\" is invalid in /etc/nginx/nginx.conf:208\n","stream":"stderr","time":"2024-09-12T09:06:24.354199756Z"}
看一下nginx.conf 280行的配置内容。

根据csghub-server.log内容看，像是docker-compose启动后，解析不了通过docker-compse.yaml文件中定义的服务名。

是通过docker-compse up命令启动的吗？

**Reply by kyrieirvingcsgo** [reply]
> nginx的log报错：{"log":"nginx: [emerg] server name "*." is invalid in /etc/nginx/nginx.conf:208\n","stream":"stderr","time":"2024-09-12T09:06:24.354199756Z"} 看一下nginx.conf 280行的配置内容。
> 
> 根据csghub-server.log内容看，像是docker-compose启动后，解析不了通过docker-compse.yaml文件中定义的服务名。
> 
> 是通过docker-compse up命令启动的吗？

是的，先执行了docker-compose up -d  然后 用的.startup.sh  启动

上面那个问题已经解决，server name "*." 改成server_name *.opencsg.public; 就好了

**Reply by kyrieirvingcsgo** [reply]
> nginx的log报错：{"log":"nginx: [emerg] server name "*." is invalid in /etc/nginx/nginx.conf:208\n","stream":"stderr","time":"2024-09-12T09:06:24.354199756Z"} 看一下nginx.conf 280行的配置内容。
> 
> 根据csghub-server.log内容看，像是docker-compose启动后，解析不了通过docker-compse.yaml文件中定义的服务名。
> 
> 是通过docker-compse up命令启动的吗？

你好，现在还有个问题

nginx容器无法启动
提示
![image](https://github.com/user-attachments/assets/5a32df77-7fbf-48a2-b1f4-38a7de06f9ca)

**Reply by Rader** [reply]
参考这个检查 nginx 的配置：https://stackoverflow.com/questions/42720618/nginx-show-to-resolve-stopped-emerg-11-host-not-found-in-upstream
